Assets/Scripts/MiniGames/Song/SongGameController.cs

```diff
@@ -2,47 +2,122 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.UI;
+using System.IO;
+using System.Linq;
 
 public class SongGameController : MonoBehaviour
 {
-    public AudioSource[] musicList;
-    private int songChoiceNum;
-
+	public static bool party_song=false, War_song=false, Sorrow_Song=false, Wisdom_Song=true;
+	public Text victory, failure;
+	private AudioSource[] musicList;
+	public AudioSource[] party_list, War_list, Sorrow_list, Wisdom_list;
+	private int songChoiceNum;
+	public Animator am;
     public static bool startPlaying;
-
-    //End Game Manager
-    public static float targetScore;
+	public CanvasGroup win, lose;
+	//End Game Manager
+	public static float targetScore;
     public float gameEndTimerValue;
     private float gameEndTimerHolder;
     public static bool endGameState;
 
     public ArrowController arrowController;
+	private List<string> lose_message = new List<string>();
+	private List<string> lose_message_badly = new List<string>();
+	private List<string> did_not_win_message = new List<string>();
+	private List<string> Win_message = new List<string>();
+	bool setmessage;
+	//score holders
+	public static float currentScore;
```

Comment:
is there any risk of all these static variables not getting reset when you restart the game? it seems like they should be member variables instead of static to me, which persist only for a single song? i might not remember the intended design though